spec/app_spec.rb

```diff
@@ -185,178 +185,93 @@
+          headers: {'Authorization' => "Bearer #{bearer_token}" },
+        ) { connection }
 
-      it 'renders expected LOA3 data when redaction is not enabled' do
-        # disable redaction
-        expect_any_instance_of(LoginGov::OidcSinatra::Config).to receive(:redact_ssn?).at_least(:once).and_return(false)
+        expect(connection).to receive(:get).with('') { Faraday::Response.new({body: { email: email}.to_json }) }
+      end
 
+      it 'takes an authorization code and gets a token, and renders the email from the token' do
         get '/auth/result', code: code
-        follow_redirect!
 
-        expect(last_response.body).to include('012-34-5678')
-        expect(last_response.body).to include('0125551212')
-        expect(last_response.body).to include('LOA3')
-        expect(last_response.body).to include('123 Main St., Anytown, US 12345')
+        expect(last_response).to be_redirect
+        follow_redirect!
+        expect(last_response.body).to include(email)
       end
 
-      it 'renders redacted SSN LOA3 data when redaction is enabled' do
-        # enable redaction
-        expect_any_instance_of(LoginGov::OidcSinatra::Config).to receive(:redact_ssn?).at_least(:once).and_return(true)
+      context 'with dangerous input' do
+        let(:email) { '<script>alert("hi")</script> mallory@bar.com' }
 
-        get '/auth/result', code: code
-        follow_redirect!
+        it 'escapes dangerous HTML' do
+          get '/auth/result', code: code
+          follow_redirect!
 
-        expect(last_response.body).to_not include('012-34-5678')
-        expect(last_response.body).to include('###-##-####')
-        expect(last_response.body).to include('0125551212')
-        expect(last_response.body).to include('LOA3')
-        expect(last_response.body).to include('123 Main St., Anytown, US 12345')
+          expect(last_response.body).to_not include(email)
+          expect(last_response.body).to include('&lt;script&gt;alert(&quot;hi&quot;)&lt;/script&gt; mallory@bar.com')
+        end
       end
-    end
-
-    context 'LOA3 /auth/result without SSN' do
-      let(:id_token) {
-        JWT.encode(
-          {
-            email: email,
-            acr: 'http://idmanagement.gov/ns/assurance/loa/3',
-            phone: '0125551212',
-            address: '123 Main St., Anytown, US 12345',
-          },
-          idp_private_key,
-          'RS256',
-        )
-      }
-
-      it 'handles nil SSN when redaction is enabled' do
-        # enable redaction
-        expect_any_instance_of(LoginGov::OidcSinatra::Config).to receive(:redact_ssn?).at_least(:once).and_return(true)
 
+      it 'has a logout link back to the SP-initiated logout URL' do
         get '/auth/result', code: code
         follow_redirect!
+        doc = Nokogiri::HTML(last_response.body)
+
+        logout_link = doc.at_xpath("//div[@class='sign-in-wrap']/a[text()='\n              Log out\n            ']")
```

Comment:
I think we can use xpath contains() to worry less about the space characters?
```suggestion
        logout_link = doc.at_xpath("//div[@class='sign-in-wrap']/a[contains(text(), 'Log out')]")
```
Or possibly is there a CSS selector that might be clearer?
```suggestion
        logout_link = doc.at_css("div.sign-in-wrap a:contains('Log out')")
```

Reply:
i didn't write these tests, i just cleaned up the mocking behavior and reordered them so that the before block made more sense. i can update them if you feel strongly about it, though

Reply:
oops didn't catch that! If either of the above suggestions works easily, I would def appreciate it for future us!

Reply:
of course!

Reply:
fixed!